src/aliceVision/matching/ArrayMatcher_hnswlib.hpp

```diff
@@ -0,0 +1,187 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2020 AliceVision contributors.
+// Copyright (c) 2019 openMVG contributors.
+// Copyright (c) 2019 Romain Janvier and Pierre Moulon
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "aliceVision/matching/ArrayMatcher.hpp"
+#include "aliceVision/matching/CascadeHasher.hpp"
+#include "aliceVision/matching/IndMatch.hpp"
+
+#include <aliceVision/system/Logger.hpp>
+
+#include "dependencies/hnswlib/hnswlib/hnswlib.h"
+
+#include <memory>
+
+#if ALICEVISION_IS_DEFINED(ALICEVISION_HAVE_OPENMP)
+#include <omp.h>
+#endif
+
+#include <vector>
+
+#include <random>
+#include <cmath>
+
+using hnswlib::HierarchicalNSW;
+using hnswlib::L2Space;
+using hnswlib::L2SpaceI;
+using hnswlib::SpaceInterface;
+using namespace std;
```

Comment:
Just to clarify you're saying to remove only `using namespace std;` and not line 31 to 33 that you quoted right?

Reply:
yes remove L34.

And actually it would be better to move those other `using` inside the class, after `using  DistanceType = typename Metric::ResultType;` for example

Reply:
For moving L31 - L33 inside the class after the line `using DistanceType = typename Metric::ResultType;` I am getting an error `Using declaration in class refers into 'hnswlib::', which is not a class`, which I think has to do with scope. I tried looking for answers online but I wasn't able to find anything. Do I need to each of the lines somehow to look more like the line before `using DistanceType = typename Metric::ResultType;`?

Reply:
ah yes my bad, it's not a derived class. You can leave them like that.